syntax="proto3";
package test.protobuf;
option cc_generic_services=true;
message Result{
    int32 code=1;
    bytes msg=2;
}
message LoginRequest{
    bytes name=1;
    bytes pwd=2;
}
message LoginReponse{
    Result result=1;
    bool success=2;
}

message User{
    int32 id=1;
    bytes name=2;
    uint32 age=3;
    enum Sex{
        UNKNOWN=0;
        MALE=1;
        FEMALE=2;
    }
    Sex sex=4;
}

message GetFriendListRequest{
    Result result=1;
}
message GetFriendListResponse{
    Result result=1;
    repeated User user=2;
}

service UserService{
    rpc Login(LoginRequest) returns (LoginReponse);
    rpc GetFriendList(GetFriendListRequest) returns (GetFriendListResponse);
}